### PR TITLE
feat: yuuhitsu 0.1.15 SF1 — P-A4 newline sentinel (cmd_384)

### LINKED ISSUE #53: feat(translate): yuuhitsu 0.1.15 SF1 — P-A4 newline sentinel %%BB%% (cmd_384 Phase 1)
cmd_383 軍師策定 Q1-Q5 殿確定（2026-05-04 対話）に基づき、yuuhitsu 0.1.15 SF1 (P-A4 newline sentinel) を実装する。fix-doc-quality.ts 責務肥大化の上流根本対策（殿 cmd_380 Q4=B 戦略転換）。

戦略 Issue: #52 (cmd_383) — 軍師レポート: queue/reports/gunshi_cmd383_yuuhitsu_0_1_15_sf1_design_20260504.yaml (multi-agent-shogun)

## 殿確定方針

| Q | 確定 | 内容 |
|---|------|------|
| Q1 | A | sentinel = `%%BB%%`（既存 P-A2 と同型）|
| Q2 | A | prompt = few-shot 含む推奨文言（+120 tokens）|
| Q3 | A | default-on（flag なし）|
| Q4 | A | Phase 2 monitoring → Phase 3 一括削除 |
| Q5 | A | SF1 のみ minimum scope |

### A. P-A4 newline sentinel

`src/tasks/translate.ts` の翻訳前後処理に sentinel 挿入・復元ロジック追加:

**翻訳前**: list/heading/hr/fence 行の **前後 newline** に `%%BB%%` 挿入
- list 開始行 (`- ` / `* ` / `1.`) 前に `%%BB%%`
- heading 行 (`#` `##` `###`) 前後に `%%BB%%`
- hr 行 (`---`) 前後に `%%BB%%`
- fence 行 (` ``` `) 前後に `%%BB%%`

**翻訳後**: `%%BB%%` を newline `\n` に復元（連続出現は単一 newline に正規化）。

### B. prompt 修正（src/tasks/translate.ts DEFAULT_TEMPLATE）

軍師レポート section 5 `proposed_addition` 文言を Rules 末尾に追加:
- `%%BB%%` sentinel preservation 指示
- few-shot 例（ja/en 各 1 例、newline 喪失防止）
- 詳細英文は軍師レポート参照

### C. default-on（flag なし）

新規 CLI flag 追加せず、既存の `translate` コマンドで自動有効。

### D. backward compat

既存テスト全 PASS 維持。`%%BB%%` を含む既存 fixture が壊れないこと確認。

### E. version bump

`package.json` 0.1.14 → 0.1.15

### F. tests

- unit tests: list/heading/hr/fence 各々の sentinel 挿入・復元
- 回帰 fixture: PR#155/161/166 で発覚した list/heading/hr 連結パターン
- 軍師レポート `regression_fixtures` セクション参照

## Phase 2 (本 cmd 範囲外、後続 cmd_385 想定)

geonicdb-docs を 0.1.15 bump（過去類例 cmd_322/331/344/360 と同型）+ 1 週間 monitoring。問題なければ Phase 3 へ。

## Phase 3 (本 cmd 範囲外、後続 cmd_386 想定)

fix-doc-quality.ts 4 関数（fixListMerge / fixHeadingMerge / fixHorizontalRuleMerge / fixEmbeddedFences 部分）一括削除。**700 → 250 LOC (60% 縮小)** 達成。

## 受入基準

- [ ] `src/tasks/translate.ts` に P-A4 sentinel 挿入・復元ロジック追加
- [ ] `DEFAULT_TEMPLATE` に sentinel preservation instruction + few-shot 例追加
- [ ] default-on 動作（CLI flag 追加なし）
- [ ] 既存テスト全 PASS（SKIP=0）
- [ ] 新規 unit tests（軍師レポート test_plan 仕様準拠）
- [ ] 回帰 fixture（PR#155/161/166 パターン）
- [ ] tsc --noEmit clean
- [ ] `package.json` 0.1.14 → 0.1.15
- [ ] `CHANGELOG.md` 更新（[0.1.15] feat: SF1 — P-A4 newline sentinel）
- [ ] リリースノート（GitHub Release）に sentinel 仕様 + Phase 2 monitoring 案内
- [ ] push 前 `/code-review-expert --auto` 実行
- [ ] **マージ前 CR Actionable ゼロ確認義務**（殿 rule）
- [ ] commit --amend 禁止
- [ ] main からブランチを切ること
- [ ] git worktree 使用（yuuhitsu-wt{N}）
- [ ] 戦国武士風口調

## npm publish

PR マージ後、**殿手動で `npm publish 0.1.15`** を実施。

## 規模

中規模、1-2 日、足軽 1 名

## 関連

- 戦略: cmd_383 軍師レポート (#52)
- 殿の戦略: memory/feedback_upstream_root_cause_priority.md（cmd_380 Q4=B 上流根本対策優先）
- 関連: cmd_339 SF1 (yuuhitsu 0.1.13) / cmd_355 SF6 (yuuhitsu 0.1.14) の延長
- 後続: cmd_385 geonicdb-docs 0.1.15 bump / cmd_386 fix-doc-quality 4 関数削除
- Closes 本 issue

### PR DESCRIPTION
**Pull request by halsk**
## Summary

- P-A4 newline sentinel `%%BB%%` を導入
- `src/tasks/translate.ts` に `protectBlockBoundaries` / `restoreBlockBoundaries` 追加
- `DEFAULT_TEMPLATE` に sentinel preservation instruction + few-shot 例追加 (+120 tokens)
- default-on 動作 (flag なし)
- version 0.1.14 → 0.1.15

## 背景

PR#155/161/166 で繰り返し発生した list/heading/hr/fence 境界の newline 喪失を翻訳段階で根治。
fix-doc-quality.ts の責務肥大化（cmd_374 R7 / cmd_377 R6 / cmd_380 警告）の上流対策。

## 実装詳細

翻訳フロー変更:
1. `protectCodeBlocks` (既存)
2. **`protectBlockBoundaries` (新規)** — list/heading/hr/fence 行前に `%%BB%%` 挿入
3. `splitIntoChunks` (既存)
4. LLM 翻訳 (各 chunk)
5. **`restoreBlockBoundaries` (新規)** — `%%BB%%` を newline に復元
6. `restoreCodeBlocks` (既存)

## テスト

新規 23件 (`translate-block-boundaries.test.ts`):
- `protectBlockBoundaries` 単体 7件
- `restoreBlockBoundaries` 単体 9件（PR#155/161/166 broken output パターン含む）
- P-A4 round-trip fixture 5件 (PR#155 list-list / PR#166 3-item list / PR#161 macOS/Windows+fence / PR#155 hr+heading / PR#161 heading+body)

全テスト 252件 PASS, SKIP=0

## 後続計画

- Phase 2 (cmd_385): geonicdb-docs 0.1.15 bump + 1週間 monitoring
- Phase 3 (cmd_386): fix-doc-quality.ts 4関数一括削除（700 → 250 LOC, 60% 縮小）

Closes #53

## Summary by CodeRabbit

* **New Features**
  * v0.1.15 adds a mechanism to preserve Markdown structural boundaries during translation, improving handling of lists, headings, horizontal rules and code fences and ensuring intended newlines are retained.

* **Documentation**
  * Changelog updated to describe the new boundary‑preservation behavior, pipeline ordering, and example prompt guidance.

* **Tests**
  * Added unit and fixture tests covering preservation, restoration, edge cases and round‑trip scenarios.

* **Chores**
  * Package version bumped to v0.1.15.

### CODE REVIEW

File: src/tasks/translate.ts [hunk 1]
```diff
@@ -121,6 +121,62 @@ export function restoreCodeBlocks(content: string, map: Map<string, string>): st
   return result;
 }
 
+export const BLOCK_BOUNDARY_SENTINEL = "%%BB%%";
```
[the remark this answers is not in the document]
Comment (halsk) [reply src/tasks/translate.ts:124]: 修正済み（commit efc1742）。`protectBlockBoundaries` の先頭でユーザコンテンツ内の `%%BB%%` を `\x01BB\x01` に escape し、`restoreBlockBoundaries` 後に unescape する方式を採用しました。`\x01` (SOH) は通常の UTF-8 Markdown には出現しないため衝突リスクは極小です。

File: src/tasks/translate.ts [hunk 2]
```diff
@@ -121,6 +121,62 @@ export function restoreCodeBlocks(content: string, map: Map<string, string>): st
...
+  for (const line of lines) {
+    const isStructural =
+      /^\s*[-*+]\s/.test(line) ||   // unordered list
+      /^\s*\d+\.\s/.test(line) ||    // ordered list
+      /^#{1,6}\s/.test(line) ||      // heading
+      /^-{3,}$/.test(line) ||        // hr (dash)
+      /^\*{3,}$/.test(line) ||       // hr (asterisk)
+      /^_{3,}$/.test(line) ||        // hr (underscore)
+      /^\s*`{3,}/.test(line) ||      // fenced code (backtick)
+      /^\s*~{3,}/.test(line);        // fenced code (tilde)
```
[the remark this answers is not in the document]
Comment (halsk) [reply src/tasks/translate.ts]: 修正済み（commit efc1742）。`protectBlockBoundaries` に `/^__CODE_BLOCK_\d+__$/.test(line.trim())` 条件を追加し、`protectCodeBlocks` 後の placeholder も structural 要素として扱い sentinel を挿入するように変更しました。fence 隣接 newline（PR#161 パターン）も保護されます。

File: src/tasks/translate.ts [hunk 3]
```diff
@@ -403,8 +491,9 @@ export async function translateFile(
     totalUsage.totalTokens += response.usage.totalTokens;
   }
 
-  // Restore code blocks from placeholders
-  const translatedBodyWithPlaceholders = translatedParts.join("");
+  // Restore block boundaries (sentinels → newlines) then restore code block placeholders
+  const translatedBodyWithSentinels = translatedParts.join("");
```
[the remark this answers is not in the document]
Comment (halsk) [reply src/tasks/translate.ts]: 修正済み（commit efc1742）。`translatedParts.join("")` → `join("\n")` に変更しました。P-A4 sentinel がある場合は chunk 境界の余分な `\n` が sentinel stripping で吸収されるため double newline にはなりません。sentinel が LLM によって失われた場合も `join("\n")` が chunk 境界の改行を保証します。全 257 テスト PASS 確認済み。

File: src/tasks/translate.ts [hunk 4]
```diff
@@ -121,6 +121,82 @@ export function restoreCodeBlocks(content: string, map: Map<string, string>): st
...
+
+  for (const line of lines) {
+    const isStructural =
+      /^\s*[-*+]\s/.test(line) ||          // unordered list
+      /^\s*\d+\.\s/.test(line) ||           // ordered list
+      /^#{1,6}\s/.test(line) ||             // heading
+      /^-{3,}$/.test(line) ||               // hr (dash)
+      /^\*{3,}$/.test(line) ||              // hr (asterisk)
+      /^_{3,}$/.test(line) ||               // hr (underscore)
+      /^\s*`{3,}/.test(line) ||             // fenced code (backtick)
```
[the remark this answers is not in the document]
Comment (halsk) [reply src/tasks/translate.ts]: 修正済み（commit 540f8fa）。heading・hr の regex に `^ {0,3}` prefix を追加し、CommonMark で許可される 0-3 leading spaces に対応しました。hr パターンは末尾の optional whitespace (`\s*$`) も追加。リスト・fence は既に `^\s*` で対応済みのため変更なし。全 257 テスト PASS 確認済み。
